### ISSUE TITLE
Native AMP & Auto Adsense do not work on desktop version

### ISSUE THREAD

**Issue by markbiernat** [ampproject/amphtml#16599]
I use AMP by Wordpress to natively display AMP pages.

" Native -  Reuses active theme's templates to display AMP responses but does not use separate URLs for AMP. Your canonical URLs are AMP. AMP-specific blocks are available for inserting into content. Any AMP validation errors are automatically sanitized."

I use auto AMP Adsense ads.
I also have the  normal auto Adsense ads script in my header just in case. (I have placed and remove this code with no effect, while always keeping the Auto AMP for Adsense code in the appropriate place).

I see ads on my mobile version. 
However,
I do not see ads on my desktop version.

However, if I turn on Adblock in my broswer (tested on Firefox and Chrome) on I see Adblock blocking ads on my desktop version, however, on or off never displaying ads except on my mobile device.

The WP AMP plugin creator thinks it is not a plugin issue but an amproject or even adsense issue.

**Comment by lannka** [comment]
The error means the ad is not placed because the page run out of possible placement slots, which is defined by the strategy.
 
@tlong2 can you take a look if it's a problem should be fixed by pub configuration?

**Comment by tlong2** [comment]
The "no anchor element found" and "unable to fulfill ad strategy" are just debug information.

The auto ads config returns a bunch of CSS selectors that describe elements on the page that ads could be inserted next to. Not every element will exist on every page and in this case "no anchor element found" is outputted in the console.

There are a maximum number of ads that amp auto ads will ever place on a page. The maximum number would likely only get inserted on a very long page (since amp auto ads also ensures that ads don't appear too close to each other). When this max isn't reached "unable to fulfill ad strategy" gets outputted in the console.

So these messages aren't anything to worry about in themselves, but can be useful if debugging amp auto ads to figure out a problem that might be occurring with it on a site.

**Comment by lannka** [comment]
@tlong2 could you answer address @markbiernat question about if auto-ads are supported in desktop? 

Also, to reduce confusion can we re-phrase those messages?

**Comment by lannka** [comment]
@markbiernat thanks for the debugging info. 

I played with @jamescridland 's site, and I found ads **below the fold** are showing up.

However, due to "no UI reflow" policy, all the ads **above the fold** are not being expanded, and ended up with 0px height.

Here is an **above-the-fold** ad with 0px height:

<img width="500" alt="screen shot 2018-07-16 at 11 10 32 am" src="https://user-images.githubusercontent.com/8496897/42775342-f97a25f6-88e8-11e8-8230-20b79b9753d1.png">

--------------------
The one **below-the-fold**  are showing up:

<img width="500" alt="screen shot 2018-07-16 at 11 11 01 am" src="https://user-images.githubusercontent.com/8496897/42775358-00e16d9a-88e9-11e8-9d06-fd3e7b8f91c7.png">

**Comment by markbiernat** [comment]
Thank you very much. I think proper integration with Adsense is an important part for AMP wider adoption.

OK I see what you mean in the above example. However, for whatever reason I do not see the same situation on my site: [Claritaslux](https://claritaslux.com/how-to-be-romantic/)

In fact if I view source I do not see ad-amp showing up, unless I disable the plugin.

**Comment by markbiernat** [comment]
@jamescridland First, thank you very much for the reply.
I personally can live without adsense on my desktop version, however, I think you will get similar questions in the future from others. I am beta testing the AMP Native option which I think is the future.

I have to respectfully disagree with you.  Are you sure you used a desktop, rather than a mobile device? On my mobile devices all ads display.  However, on my desktop it does not. Even when I scroll down, even when I use a VPN, I have checked it on Chrome, Firefox and Opera. 

I am using AMP in a Native option, however, paired and legacy URLs work but Native is doing something different of course.

The second I disable AMP the ads appear.

You are correct that I have validation errors. In Native mode "All validation errors are forcibly accepted when in native mode." Do you think these errors are what prevents ads from being displayed?

The errors are from the most plain vanilla install of Wordpress plugins and theme. The errors seem to relate to such things as Jetpack, googletagmanage, fonts.googleapis.com etc, which are all pretty standard tools on any site.

Again thank you.

**Comment by jamescridland** [comment]
@markbiernat Here's what I see. I have no idea what you mean by 'native mode'. The only error that the AMP validator extension kicks up is that you have TWO amp-auto-ads extensions loading. I'd fix that, if I were you.

![claritas](https://user-images.githubusercontent.com/231941/42790340-e4e96762-89ad-11e8-96c7-065eb0b502c7.png)

**Comment by markbiernat** [comment]
Here is what is ment by Native:
![native](https://user-images.githubusercontent.com/30418497/42793064-50f1b6f2-8946-11e8-90c5-a64f27ea8755.jpg)
Its a plugin that Wordpress/Automattic creates
![plugin](https://user-images.githubusercontent.com/30418497/42793084-6414f35c-8946-11e8-8617-b38116125a09.jpg)

**Comment by jamescridland** [comment]
OK, I've no idea about Wordpress, and this isn't a wordpress project.

**Comment by markbiernat** [comment]
@jamescridland 
Yes agreed and sorry for the confusion, I would not just randomly post on the wrong forum. This is not a third party plugin but rather created by Wordpress itself in cooperation from my understanding with Google and AMP. 
And when initially looked at on the WP support forum, the creator of this for Wordpress AMP said, Adsense not showing on the desktop view, was not a WP issue but rather might be a AMP Project issue?  That is why I posted here. However, if you are saying it is not related to an AMP issue than it might be an Google Adsense issue.

**Comment by markbiernat** [comment]
As this might be help to others, I fixed the validation error: tag 'amp-auto-ads extension .js script' appears more than once in the document."

By removing:

 "async custom-element="amp-auto-ads"
"://cdn.ampproject.org/v0/amp-auto-ads-0.1.js" - placed once in the header


 then adding this to my functions.php:
add_filter( 'wp_head', function( $data ) {
	$data['amp_component_scripts'] = array_merge(
		$data['amp_component_scripts'],
		array(
			'amp-ad' => 'https://cdn.ampproject.org/v0/amp-ad-latest.js',
		)
	);
	return $data;
} );

This works and might help others, but this is Wordpress, rather than AMP specific.

**Comment by lannka** [comment]
@markbiernat sorry for the frustration here, and thanks for the tips to fix the validation error.
Did you get ads back on desktop mode after this? Fixing validation error is important, since only valid AMP pages will be cached. 

@jamescridland thank you very much for the help here. Given more sites started to do "Native AMP" (what we normally call canonical AMP), I think we should start to think what different UX policies on desktop. For example, is it still that important to disallow element insertion/resize on the first viewport.

**Comment by lannka** [comment]
@markbiernat FYI, the auto ads not working on IE Firefox has been fixed #16883

Regarding to the validation error, were you following the guidelines [here](https://support.google.com/adsense/answer/9011465) from AdSense? I noticed your "fix" switched `amp-auto-ads-0.1.js` to `amp-ad-latest.js`, did you actually mean `amp-auto-ads-latest.js`. But even that I don't think matters. Could you send over a snippet of your invalid AMP? Or try it on https://validator.ampproject.org/

@morsssss has experience with the wordpress plugin, could you take a look? 
tldr; @markbiernat is using wordpress plugin to generate a page with amp-auto-ads extension, and if "Native AMP" is turned on  the output didn't pass validation.

**Comment by markbiernat** [comment]
@lannka here is an example of my validation errors on my polishgrammar.com site.   My sites all have the same set up, basic Twenty Seventeen theme and standard WP Jetpack.   When I use the Wordpress sponsored AMP plugin in Native mode  it tell me: "Any AMP validation errors are automatically sanitized. " and ..."All validation errors are forcibly accepted when in native mode"
But my dashboard still is aware of the errors.
Here is a text file of the errors, because it is too long to copy and paste.
[validation errors.txt](https://github.com/ampproject/amphtml/files/2271762/validation.errors.txt)
[invalid pages.txt](https://github.com/ampproject/amphtml/files/2271763/invalid.pages.txt)

**Comment by lannka** [comment]
@markbiernat sorry I don't understand those error messages in the files you provided. It seems to be errors from the plugin not the AMP validators. We might need help from the maintainer of the plugin to shed a light on this.

**Comment by lannka** [comment]
@markbiernat given this thread was started with something different, which should be already resolved (the confusing error messages and auto ads not working on FF). 

To let any new participants here quickly get to the point, would you please open a separate issue just for the AMP validation errors in the plugin?
